src/Development/IDE/Plugin/CodeAction.hs

```diff
@@ -181,6 +182,17 @@ suggestAction dflags packageExports ideOptions parsedModule text diag = concat
     ] ++
     suggestFillHole diag                   -- Lowest priority
 
+suggestDisableWarning :: Diagnostic -> [(T.Text, [TextEdit])]
+suggestDisableWarning Diagnostic{..}
+    | Just (StringValue (showFlag -> Just w)) <- _code =
+        pure
+            ( "Disable \"" <> w <> "\" warnings"
+            , [TextEdit (Range (Position 0 0) (Position 0 0)) $ "{-# OPTIONS_GHC -Wno-" <> w <> " #-}\n"]
+            )
+    | otherwise = []
+  where
+    showFlag = fmap camelToHyphenCase . T.stripPrefix "Opt_Warn"
+    camelToHyphenCase = T.dropWhile (== '-') . T.concatMap (\c -> if isUpper c then "-" <> T.singleton (toLower c) else T.singleton c)
```

Comment:
We need to go from e.g. "Opt_WarnUnusedImports" to "unused-imports". I'd expect this to be in the GHC API somewhere, but I can't find anything.

Reply:
There is `fmap flagName . findSpecOf`.

Reply:
Where does `findSpecOf` come from? I can't find that anywhere.

Reply:
Sorry, I meant `flagSpecOf`: https://hackage.haskell.org/package/ghc-8.10.1/docs/src/DynFlags.html#flagSpecOf

But it turns out it's not imported :(

Reply:
You mean not exported? Perhaps we should ask GHC to change that.

Turns out this logic is only valid for 68 out of 82 warnings...

Reply:
Closed by 71da233

Reply:
> Closed by 71da233

Actually resolved in a more maintainable way by 5a2ffa4.